### PR TITLE
feat: availability in instant meeting 

### LINKED ISSUE #16430: [CAL-4215] Add availability dropdown in instant meeting settings
* enable instant meeting
* add a dropdown to select any availability
* on the booking page we would only check if current time is between the selected availability of instant meeting and show modal.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4215](https://linear.app/calcom/issue/CAL-4215/add-availability-dropdown-in-instant-meeting-settings)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

TODO:-

- [x] Unit Tests

- Fixes  https://github.com/calcom/cal.com/issues/16430 (GitHub issue number)
- Fixes CAL-4215 (Linear issue number - should be visible at the bottom of the GitHub issue description)

https://github.com/user-attachments/assets/8d8d36dd-b827-4c63-82d9-f9fac4531158

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1) Create a org team event type
2) Enable instant booking in event type settings
3) select any availability from dropdown and click save 
4) Now check the booking page and "Connect Now" modal should be only visible when current time is between the availability selected.

### CODE REVIEW

File: packages/features/eventtypes/lib/getPublicEvent.ts [hunk 1]
```diff
@@ -120,11 +121,79 @@ const publicEventSelect = Prisma.validator<Prisma.EventTypeSelect>()({
...
+      timeZone: true,
+    },
+  },
+
   hidden: true,
   assignAllTeamMembers: true,
   rescheduleWithSameRoundRobinHost: true,
 });
 
+export async function isCurrentlyAvailable({
```
Comment (Udit-takkar) [inline packages/features/eventtypes/lib/getPublicEvent.ts:136]: This function is just to check if current time is between the available time. It doesn't check if users are already booked, calendar blocked, OOO. 

As this is just for instant booking event where user can book for future slots if no one is available currently

File: packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx [hunk 2]
```diff
@@ -42,13 +76,30 @@ export default function InstantEventController({
...
+  const schedules = data.schedules;
+
+  const options = schedules.map((schedule) => ({
+    value: schedule.id,
+    label: schedule.name,
+    isDefault: schedule.isDefault,
+    isManaged: false,
+  }));
+
+  console.log("options", options);
```
Comment (ThyMinimalDev) [inline packages/features/eventtypes/components/tabs/instant/InstantEventController.tsx:102]: log to remove

**Review by ThyMinimalDev — COMMENTED** [review]
well done, looking good, only one console log to remove and we can merge

**Review by alishaz-polymath — APPROVED** [review]
LGTM 🚀

**Pull request merged**